protocols/v2/codec-sv2/src/encoder.rs

```diff
@@ -1,3 +1,26 @@
+//! # Encoder
+//!
+//! Provides utilities for encoding messages into Sv2 frames, with or without Noise protocol
+//! support.
+//!
+//! ## Usage
+//!
+//! All messages passed between Sv2 roles are encoded as Sv2 frames using primitives in this
+//! module. There are two types of encoders for creating these frames: one for regular Sv2 frames
+//! (`Encoder`), and another for Noise-encrypted frames (`NoiseEncoder`). Both encoders manage the
+//! serialization of outgoing data and, when applicable, the encryption of the data before
+//! transmission.
+//!
+//! ### Buffer Management
+//!
+//! The encoders rely on buffers to hold intermediate data during the encoding process.
+//!
+//! - When the `with_buffer_pool` feature is enabled, the internal `Buffer` type is backed by a
+//!   pool-allocated buffer (`binary_sv2::BufferPool`), providing more efficient memory usage,
+//!   particularly in high-throughput scenarios.
+//! - If the feature is not enabled, a system memory buffer (`binary_sv2::BufferFromSystemMemory`)
+//!   is used for simpler applications where memory efficiency is less critical.
```

Comment:
the module is private, which means this header is for maintainer eyes only

whenever we make the module public we can change this, but for now let's document things as they are.

```suggestion
// # Encoder
//
// Provides utilities for encoding messages into Sv2 frames, with or without Noise protocol
// support.
//
// ## Usage
//
// All messages passed between Sv2 roles are encoded as Sv2 frames using primitives in this
// module. There are two types of encoders for creating these frames: one for regular Sv2 frames
// (`Encoder`), and another for Noise-encrypted frames (`NoiseEncoder`). Both encoders manage the
// serialization of outgoing data and, when applicable, the encryption of the data before
// transmission.
//
// ### Buffer Management
//
// The encoders rely on buffers to hold intermediate data during the encoding process.
//
// - When the `with_buffer_pool` feature is enabled, the internal `Buffer` type is backed by a
//   pool-allocated buffer (`binary_sv2::BufferPool`), providing more efficient memory usage,
//   particularly in high-throughput scenarios.
// - If the feature is not enabled, a system memory buffer (`binary_sv2::BufferFromSystemMemory`)
//   is used for simpler applications where memory efficiency is less critical.
```